core/iwasm/aot/aot_runtime.c

```diff
@@ -1210,6 +1240,133 @@ aot_instantiate(AOTModule *module, AOTModuleInstance *parent,
     }
 #endif
 
+#if WASM_ENABLE_GC != 0
+    if (!is_sub_inst) {
+        uint32 gc_heap_size = wasm_runtime_get_gc_heap_size_default();
+        AOTModuleInstanceExtra *extra =
+            (AOTModuleInstanceExtra *)module_inst->e;
+
+        if (gc_heap_size < GC_HEAP_SIZE_MIN)
+            gc_heap_size = GC_HEAP_SIZE_MIN;
+        if (gc_heap_size > GC_HEAP_SIZE_MAX)
+            gc_heap_size = GC_HEAP_SIZE_MAX;
+
+        extra->gc_heap_pool =
+            runtime_malloc(gc_heap_size, error_buf, error_buf_size);
+        if (!extra->gc_heap_pool)
+            goto fail;
+
+        extra->gc_heap_handle =
+            mem_allocator_create(extra->gc_heap_pool, gc_heap_size);
+        if (!extra->gc_heap_handle)
+            goto fail;
+    }
+#endif
+
+#if WASM_ENABLE_GC != 0
+    /* Initialize the table data with table init data */
```

Comment:
Not sure whether it is easy to move these codes into the end of `tables_instantiate`?

Reply:
Some info needed by these logic is not ready in `tables_instantiate`, such as function pointer and function type (init_func_ptrs/init_func_type_indexes is called after the tables_instantiate).

Reply:
OK, got it.